pkg/registry/generic/etcd/etcd.go

```diff
@@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 		if newVersion != version {
 			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the resource was updated to %d", version))
+			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("another client made an update to this object in resource version %d", version))
```

Comment:
----- Original Message -----
> > @@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj
> > runtime.Object) (runtime.Object, bool
> >  		}
> >  		if newVersion != version {
> >  			// TODO: return the most recent version to a client?
> > -			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("the resource was updated to %d", version))
> > +			return nil, nil, kubeerr.NewConflict(e.EndpointName, name,
> > fmt.Errorf("another client made an update to this object in resource
> > version %d", version))
> 
> I don't think this text is a net win. How about something more like one of
> these:
> 
> * "the resource has been changed since you read it; re-read it and try your
> change again"
> * "the resource's version has been incremented to %v; please retry"
> * "changes must be made against the most recent resource version"
> * "the resourceVersion you reported (%v) is out-of-date; please re-read the
> object and try your change again"

We wanted to catch the version (useful for anyone who knows what they have to do) and that the likely cause was someone else actually changed this object (to slow someone down from doing an overwrite, or to explain to a brand new user why it failed).  The first example with resource version included would be fine for me as well.